src/app/context/appState.tsx

```diff
@@ -107,7 +107,11 @@ function actionHandler(draft: Draft<AppState>, action: AppStateAction) {
             kind,
           );
           if (measurement) {
-            draft.view.selectedMeasurements[kind] = measurement.measurement.id;
+            const id = measurement.measurement.id;
+            draft.view.selectedMeasurements[kind] = id;
+            if (draft.view.currentMeasurement === undefined) {
+              draft.view.currentMeasurement = { id, kind };
+            }
```

Comment:
This doesn't sound right. It means we can have a different current measurement to the selected measurement.

Reply:
I made this part of the state in https://github.com/zakodium-oss/analysis-ui-components/commit/2df2cc06780a2fb1459d67388fda7495423a1232

I think we can change `currentMeasurement` to be `displayedKind`. The measurement displayed would be the `selectedMeasurement[displayedKind]`

@targos I suggest we create a new issue for that since the functionality for this PR seems to be ok